tests/test_operation.py

```diff
@@ -394,3 +393,148 @@ class DummyObserv(qml.operation.Observable):
+        Y = qml.Hermitian(p, wires=[1, 2], do_queue=False)
+        Z = qml.Hermitian(p, wires=[1, 2], do_queue=False)
+        t = Tensor(X, Y, Z)
+        assert t.num_params == 2
+
+    def test_parameters(self):
+        """Test that the correct nested list of parameters is returned"""
+        p = np.array([0.5])
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.Hermitian(p, wires=[1, 2], do_queue=False)
+        t = Tensor(X, Y)
+        assert t.parameters == [[], [p]]
+
+    def test_multiply_obs(self):
+        """Test that multiplying two observables
+        produces a tensor"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.Hadamard(2, do_queue=False)
+        t = X @ Y
+        assert isinstance(t, Tensor)
+        assert t.obs == [X, Y]
+
+    def test_multiply_obs_tensor(self):
+        """Test that multiplying an observable by a tensor
+        produces a tensor"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.Hadamard(2, do_queue=False)
+        Z = qml.PauliZ(1, do_queue=False)
+
+        t = X @ Y
+        t = Z @ t
+
+        assert isinstance(t, Tensor)
+        assert t.obs == [Z, X, Y]
+
+    def test_multiply_tensor_obs(self):
+        """Test that multiplying a tensor by an observable
+        produces a tensor"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.Hadamard(2, do_queue=False)
+        Z = qml.PauliZ(1, do_queue=False)
+
+        t = X @ Y
+        t = t @ Z
+
+        assert isinstance(t, Tensor)
+        assert t.obs == [X, Y, Z]
+
+    def test_multiply_tensor_tensor(self):
+        """Test that multiplying a tensor by a tensor
+        produces a tensor"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.PauliY(2, do_queue=False)
+        Z = qml.PauliZ(1, do_queue=False)
+        H = qml.Hadamard(3, do_queue=False)
+
+        t1 = X @ Y
+        t2 = Z @ H
+        t = t2 @ t1
+
+        assert isinstance(t, Tensor)
+        assert t.obs == [Z, H, X, Y]
+
+    def test_multiply_tensor_in_place(self):
+        """Test that multiplying a tensor in-place
+        produces a tensor"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.PauliY(2, do_queue=False)
+        Z = qml.PauliZ(1, do_queue=False)
+        H = qml.Hadamard(3, do_queue=False)
+
+        t = X
+        t @= Y
+        t @= Z @ H
+
+        assert isinstance(t, Tensor)
+        assert t.obs == [X, Y, Z, H]
+
+    def test_operation_multiply_invalid(self):
+        """Test that an exception is raised if an observable
+        is multiplied by an operation"""
+        X = qml.PauliX(0, do_queue=False)
+        Y = qml.CNOT(wires=[0, 1], do_queue=False)
+        Z = qml.PauliZ(0, do_queue=False)
+
+        with pytest.raises(ValueError, match="Can only perform tensor products between observables"):
+            X @ Y
+
+        with pytest.raises(ValueError, match="Can only perform tensor products between observables"):
+            T = X @ Z
+            T @ Y
```

Comment:
I would like to support this, but I think it requires some thinking/planning. It would be nice to have a `TensorSum` class, that allows you to do something like

```
O = 2 * X @ Y + a Z @ H
```
where perhaps `a` is a free parameter. It's just a matter of working out how to let the QNode know the 'classical' gradient rule `dO/da = Z @ H`.

Reply:
k we can leave for future consideration